lib/lita/handlers/reviewme.rb

```diff
@@ -116,6 +118,19 @@ def mention_reviewer(response, room: get_room(response))
         response.reply(chat_mention(reviewer, url))
       end
 
+      # Now that we lowercase inputted usernames to make our commands fully case insensitive, we
+      # need to migrate all reviewer lists to make sure all usernames are lowercase, or otherwise
+      # users would not be able to remove usernames that already had uppercase characters in them.
+      def ensure_usernames_are_lowercase(payload)
+        lists = redis.keys("*") # get all reviewme namespaced keys
```

Comment:
Sorry for the hilariously late comment here! 😓

I think what kept holding me up on this is how generalized the effect is. That is, this will have the affect of downcasing everything in the lita-reviewme namespace. Unfortunately that seems like a result of not nested "rooms" under a common key. There was nothing you could use to target the lists 😖.

I also don't love how this code has to sort of live on forever to ensure that really hold clients get correctly migrated when they upgrade. Oof. There's a part of make that wants to just make this a breaking change and let people rebuild their reviewer lists... Is there a better way?

Reply:
Hmm, not that I can think of right now. I'll try to think about this more, but I'd be happy to hear your or (anyone else's) suggestions on how to move forward here.

Reply:
I feel really crummy about holding this up for so long, but I also feel a little crummy about adding ephemeral behavior without having a clear understanding of _when_ it can go away :/. I suppose we could build a little task or something that can be run to migrate existing data? Not sure how that fits in with Lita, but maybe as simple as a rake task?

Reply:
Yeah, it'd be much cleaner if we could run a one-off migration task when the user upgraded from a previous version to the version that would contain this change.

Reply:
It's been awhile since I've managed a Lita instance with this plugin. Any intuition if such a thing is doable?